templates/compass.html

```diff
@@ -9,119 +9,160 @@ <h2>Compass<sub class="is-size-7 pl-1">find your way</sub></h2>
     </div>
     <hr>
     <div>
-        <div class="is-flex is-flex-direction-row">
-            <div style="flex:37%;text-align:left;padding:15px;">
-                <h2>Generate Layer</h2>
-                <div id="layerSelectionAdversary">
-                    <select id="layer-selection-adversary" style="margin:0 0 0 0">
-                        <option value="" selected>Select an Adversary (All)</option>
-                        {% for adv in adversaries %}
-                        <option value="{{ adv.adversary_id }}">{{ adv.name }}</option>
-                        {% endfor %}}
-                    </select>
+        <div class="is-flex is-flex-direction-row mb-4">
+            <div class="is-flex is-flex-direction-column">
+                <label for="layer-selection-adversary" class="pb-2 is-size-6">Generate Layer</label>
+                <div class="is-flex is-flex-direction-row is-align-items-center">
+                    <div id="layerSelectionAdversary" class="mt-1 mr-2">
+                        <select id="layer-selection-adversary" x-model="selectedAdversaryID">
+                            <option value="" selected>Select an Adversary (All)</option>
+                            {% for adv in adversaries %}
+                            <option value="{{ adv.adversary_id }}">{{ adv.name }}</option>
+                            {% endfor %}}
+                        </select>
+                    </div>
+                    <label for="generateLayer">
+                        <button id="generateLayer" class="button is-primary is-small"
+                                x-on:click="generateLayer()"><i class="pr-1 fas fa-download"></i>
+                            <span class="has-tooltip-multiline has-tooltip-bottom"
+                                  data-tooltip="Click 'Generate Layer'. Click '+' to open a new tab in the navigator. Select 'Open Existing Layer'. Select 'Upload from local' and upload the generated layer file.">Generate Layer
+                            </span>
+                        </button>
+                    </label>
                 </div>
-                <button id="generateLayer" type="button" class="button-success"
-                        style="" onclick="generateLayer()">Generate Layer
-                </button>
             </div>
-            <div style="flex:37%;text-align:left;padding:15px;">
-                <h2>Generate Adversary</h2>
-                <button id="generateAdversary" type="button" class="button-success"
-                        style="margin-top:-3px" onclick="uploadAdversaryLayerButtonFileUpload()">Upload
-                    Adversary
-                    Layer
-                </button>
-                <input id='adversaryLayerInput' type='file' hidden/>
+            <div class="is-flex is-flex-direction-column ml-6">
+                <label for="generateAdversary" class="pb-2 is-size-6">Generate Adversary</label>
```

Comment:
This is kind of silly, but for some reason clicking on the label text also opens the dialog to upload a file (at least in FF). Not a big deal but could be an accessibility issue down the road

Reply:
Great catch!